www/js/survey/enketo/enketo-add-note-button.js

```diff
@@ -0,0 +1,287 @@
+ * - The directive implements a `verifyTrip` function that can be invoked by
+ *      other components.
+ */
+
+angular.module('emission.survey.enketo.add-note-button',
+    ['emission.stats.clientstats',
+        'emission.services',
+        'emission.config.dynamic',
+        'emission.survey.enketo.launch',
+        'emission.survey.enketo.answer',
+        'emission.survey.enketo.preview',
+        'emission.survey.inputmatcher'])
+.directive('enketoAddNoteButton', function() {
+  return {
+    scope: {
+      trip: '=',
+      notesConfig: '=',
+      surveys: '=',
+      timeBounds: '=',
+      datakey: '@',
+    },
+    controller: "EnketoAddNoteButtonCtrl",
+    templateUrl: 'templates/survey/enketo/add-note-button.html'
+  };
+})
+.controller("EnketoAddNoteButtonCtrl", function($scope, $element, $attrs, $translate,
+    EnketoSurveyLaunch, $ionicPopover, ClientStats, DynamicConfig,
+    EnketoNotesButtonService) {
+  console.log("Invoked enketo directive controller for add-note-button");
+  $scope.notes = [];
+
+  EnketoNotesButtonService.key = $scope.datakey;
+
+  $scope.displayLabel = () => {
+    const localeCode = $translate.use();
+    // if already filled in
+    //   return $scope.notesConfig?.['filled-in-label']?.[localeCode];
+    return $scope.notesConfig?.['not-filled-in-label']?.[localeCode];
+  }
+
+  $scope.getPartialTimeUseResponse = () => {
+    const timebounds = $scope.timeBounds(); // for trips, these times are start and end of trip
+                                            // for places, it is enter and exit time (end of one trip and start of next)
+
+    // Prefill the time bounds of the trip into the survey
+    const startDayAndTime = timebounds.start_fmt_time.split('T');
+    const startHMS = startDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const startTimezoneCode = startDayAndTime[1].split('-')[1];
+    const startTime = `${startHMS}.000-${startTimezoneCode}`
+    const endDayAndTime = timebounds.end_fmt_time.split('T');
+    const endHMS = endDayAndTime[1].substring(0, 8); // truncated to 8 chars for HH:MM:SS
+    const endTimezoneCode = endDayAndTime[1].split('-')[1];
+    const endTime = `${endHMS}.000-${endTimezoneCode}`
+
+    // TODO: Hardcoding a partial TimeUseSurvey response for now
+    // Can we come up with a more generic and more elegant way to do this ?
+    return {
+      data: {
+        name: "TimeUseSurvey",
+        xmlResponse:
+        `<a88RxBtE3jwSar3cwiZTdn xmlns:jr=\"http://openrosa.org/javarosa\" xmlns:orx=\"http://openrosa.org/xforms\" id=\"a88RxBtE3jwSar3cwiZTdn\">
+          <start>${timebounds.start_fmt_time}</start>
+          <end>${timebounds.end_fmt_time}</end>
+          <group_hg4zz25>
+            <Date>${startDayAndTime[0]}</Date>    ${/* YY:MM:DD */''}
+            <Start_time>${startTime}</Start_time> ${/* HH:MM:SS.mmm-HH:MM */''}
+            <End_time>${endTime}</End_time>
```

Comment:
We basically have two options for this:
(1) we can put this into the survey config as well, or
(2) we can specify that all "notes" surveys must have these fields to work

Right now, I am tempted to go with (2) and defer (1) until we have more examples of notes surveys.
As an aside, is the date needed? wouldn't the date just be the date of the trip/place?
To answer my own question: what if somebody was at a place for multiple days? To pick a completely hypothetical example, if they were in COVID isolation, for example.

Reply:
we are going to have an xml path in the survey config
while loading the survey config, we will load the xml in addition to the json
we will parse the XML and find the tag named `<instance>`
and will use that to fill in the prefilled value

Concretely, the instance for the time use survey is as below which seems to have all the strings that we need.

```
      <instance>
        <a88RxBtE3jwSar3cwiZTdn id="a88RxBtE3jwSar3cwiZTdn">
          <start/>
          <end/>
          <group_hg4zz25>
            <Date/>
            <Start_time/>
            <End_time/>
            <Activity_Type/>
            <Personal_Care_activities/>
            <Employment_related_a_Education_activities/>
            <Domestic_activities/>
            <Recreation_and_leisure/>
            <Voluntary_work_and_care_activities/>
            <Other/>
          </group_hg4zz25>
          <meta>
            <instanceID/>
          </meta>
        </a88RxBtE3jwSar3cwiZTdn>
      </instance>
```